### ISSUE TITLE
[Feature] Triton Implementation for combined masker

### ISSUE THREAD

**Issue by xAlg-ai** [xAlg-ai/sparse-attention-hub#17]
### Describe the feature you'd like
Triton implementation of the following ResearchAttention.custom_attention:

Note: checkout hashattention-1.0 repository for hashattention weight file

    weight_file = <HashAttention-1.0>/artifacts/llama3.1-8b-patch.64K.v1.hat_weights.pkl
    sparse_attention_config_adaptive_sampling = ResearchAttentionConfig(masker_configs=[
        SinkMaskerConfig(sink_size=0.01),
        LocalMaskerConfig(window_size=0.01),
            HashAttentionTopKMaskerConfig(heavy_size=0.05, 
                                        hat_bits=32, 
                                        hat_mlp_layers=3, 
                                        hat_mlp_hidden_size=128, 
                                        hat_mlp_activation="silu",
                                        hat_weight_file=weight_file,
                                        hat_weights=None),
        AdaptiveSamplingMaskerConfig(base_rate_sampling=0.05, epsilon=0.25, delta=0.25, init_offset=0.01,    local_offset=0.01)
    ])

    custom_attention_module =    ResearchAttention.create_masker_from_config(sparse_attention_config_adaptive_sampling)

    # we want triton function that matches exactly the following:
    sample_module = nn.Module()
   **custom_attention_module.custom_attention**(
        sample_module,
        queries, # (b,h,q,d)
        keys, # ( b,h,k,d)
        values: torch.Tensor, # ( b,h,k,d)
        attention_mask: Optional[torch.Tensor], 
        scaling : float,
        dropout : dropout,
        sparse_meta_data = {},
   )

### Explanation
The custom attention function applies masks (i.e. generates idx to include) sequentially via the Masker.add_mask function
1. LocalMasker will add sliding window tokens
2. SinkMasker will add initial tokens
3. hashAttentiontopKmasker will add top-k tokens from those remaining after applying local/sink and attention_mask input.
4. adaptive sampling then does the sampling part. 

1,2 are straighforward
3 we can use existing code from HashAttention efficiency (fixed for per -head sparsity)
4 is the new thing . 

### Why is it needed?
Efficiency evaluation of this config. 

### Requiremnets 
1. Matches exact behavior of the above function.
